site/search/Autocomplete.tsx

```diff
@@ -0,0 +1,256 @@
+        ),
+        item: ({ item }) => {
+            return (
+                <div>
+                    <span>{item.title}</span>
+                </div>
+            )
+        },
+    },
+}
+
+const AlgoliaSource: AutocompleteSource<BaseItem> = {
+    sourceId: "autocomplete",
+    onSelect({ navigator, item, state }) {
+        const itemUrl = prependSubdirectoryToAlgoliaItemUrl(item)
+        navigator.navigate({ itemUrl, item, state })
+    },
+    getItemUrl({ item }) {
+        const itemUrl = prependSubdirectoryToAlgoliaItemUrl(item)
+        return itemUrl
+    },
+    getItems({ query }) {
+        return getAlgoliaResults({
+            searchClient,
+            queries: [
+                {
+                    indexName: SearchIndexName.Pages,
+                    query,
+                    params: {
+                        hitsPerPage: 2,
+                        distinct: true,
+                    },
+                },
+                {
+                    indexName: SearchIndexName.Charts,
+                    query,
+                    params: {
+                        hitsPerPage: 2,
+                        distinct: true,
+                    },
+                },
+                {
+                    indexName: SearchIndexName.Explorers,
+                    query,
+                    params: {
+                        hitsPerPage: 1,
+                        distinct: true,
+                    },
+                },
+            ],
+        })
+    },
+
+    templates: {
+        item: ({ item }) => {
+            const index = item.__autocomplete_indexName as SearchIndexName
+            const indexLabel = {
+                [SearchIndexName.Charts]: "Chart",
+                [SearchIndexName.Explorers]: "Explorer",
+                [SearchIndexName.Pages]: "Page",
+            }[index]
+
+            return (
+                <div className="aa-ItemWrapper">
+                    <span>{item.title}</span>
+                    <span>{indexLabel}</span>
+                </div>
+            )
+        },
+    },
+}
+
+const AllResultsSource: AutocompleteSource<BaseItem> = {
+    sourceId: "runSearch",
+    onSelect,
+    getItemUrl,
+    getItems({ query }) {
+        return [
+            {
+                slug: `/search?q=${encodeURI(query)}`,
+                title: `All search results for "${query}"`,
+            },
+        ]
+    },
+
+    templates: {
+        item: ({ item }) => {
+            return (
+                <div className="aa-ItemWrapper">
+                    <div className="aa-ItemContent">
+                        <div className="aa-ItemIcon">
+                            <FontAwesomeIcon icon={faSearch} />
+                        </div>
+                        <div className="aa-ItemContentBody">{item.title}</div>
+                    </div>
+                </div>
+            )
+        },
+    },
+}
+
+export const AUTOCOMPLETE_CONTAINER_ID = "#autocomplete"
```

Comment:
We'll cross that bridge when we get there but we'll probably have to adapt some of the logic which assumes that there can only be one autocomplete per page, which won't be the case on e.g. the homepage.